src/calculateRentalCost.js

```diff
@@ -4,7 +4,21 @@
  * @return {number}
  */
 function calculateRentalCost(days) {
-  // write code here
+  const rentPrice = 40;
+  const SHORT_TERM = 3;
+  const SHORT_TERM_DISCOUNT = 20;
+  const LONG_TERM = 7;
+  const LONG_TERM_DISCOUNT = 50;
+
+  if (days >= SHORT_TERM) {
+    if (days >= LONG_TERM) {
+      return days * rentPrice - LONG_TERM_DISCOUNT;
+    }
+
+    return days * rentPrice - SHORT_TERM_DISCOUNT;
+  }
```

Comment:
The current logic applies only one discount, either short-term or long-term, based on the order of conditions. However, if a rental period qualifies for a long-term discount, it should not also apply the short-term discount. Consider restructuring the conditions to ensure that only the appropriate discount is applied.